### PR TITLE
Query and update animals table.

### PR DESCRIPTION
**Pull request by Gopxfs**
# Summary :ledger:
- Added a 'species' column of type text.
- Added 6 new animals.
- Added species to all animals.
- Updated negative weights to positive values.
- Updated schema.sql, data.sql, and queries.sql files with all changes.

## Screenshots :camera:
| milestone2_vet_clinic |
| :--: |
| _Adding new animals & species_ |

| milestone2_vet_clinic_2 |
| :--: |
| _Deleting all animals and using rollback_ |

| milestone2_vet_clinic_3 1  milestone2_vet_clinic_3 2 |
| :--: |
| _Using savepoint & updating animal weights_ |

| milestone2_vet_clinic_4 |
| :--: |
|_Running all queries_|

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -8,3 +8,15 @@ select name, escape_attempts from animals where weight_kg > 10.5;
...
+select avg(weight_kg) from animals;
+select avg(escape_attempts) from animals where neutered = '0';
+select avg(escape_attempts) from animals where neutered = '1';
+select min(weight_kg) from animals where name like '%mon%';
+select max(weight_kg) from animals where name like '%mon%';
+select min(weight_kg) from animals where name not like '%mon%';
+select max(weight_kg) from animals where name not like '%mon%';
+select avg(escape_attempts) from animals where species = 'Digimon' and date_of_birth > '1990-01-01' and date_of_birth < '2000-01-01';
+select avg(escape_attempts) from animals where species = 'Pokemon' and date_of_birth > '1990-01-01' and date_of_birth < '2000-01-01';
+
```
Comment (ShahierNashaat) [inline queries.sql:22]: - [x] Kindly be noticed that you should write those queries even if you execute them on your machine.
- [x] Kindly be noticed that you should UPDATE the animals' table by setting the species column to unspecified, then roll back the change.
```sql
   BEGIN; -- start transaction

   UPDATE animals
   SET species = 'unspecified'; -- make change

   SELECT species from animals; -- verify that change was made
   ROLLBACK;

   SELECT species from animals; -- verify that change was undone
```

- [x] Also notice that you should `UPDATE` the animals' table by setting the species column to digimon for all animals that have a name ending in mon.
- [x] Also `UPDATE` the animals' table by setting the species column to pokemon for all animals that don't have species already set.
```sql
BEGIN;
   UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
   UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 

   SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
```

- [x] Also inside a transaction delete all records in the animals' table, then roll back the transaction.
```sql
BEGIN;

   DELETE FROM animals WHERE ;
   SELECT COUNT(*) FROM ANIMALS;

   ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
```

- [x] Also Inside a transaction:
* `DELETE` all animals born after Jan 1st, 2022
* Create a savepoint for the transaction
* `UPDATE` all animals' weight to be their weight multiplied by -1.
* Rollback to the savepoint
* `UPDATE` all animals' weights that are negative to be their weight multiplied by -1.
* Commit transaction
```sql
BEGIN;

    DELETE FROM animals WHERE  date_of_birth >= '2022/01/01' ;
    SAVEPOINT SP1;

    UPDATE animals SET weight_kg = weight_kg * -1; 
    ROLLBACK TO SP1;
    UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0; 

    COMMIT;
```

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi <STUDENT_MENTION>, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Good overall design ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (Gopxfs) [reply queries.sql:22]: Hey, @ShahierNashaat thank you so much for the effort, I updated the files. It wasn't clear on the requirements that I should put every action there.

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @Gopxfs, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**